### PR TITLE
Allow specifying browser arguments (closes #905)

### LINKED ISSUE #905: Allow flags for browsers
Feature Request
### What is the current behavior?

Flags are not passed to the opening browser.
### What is the expected behavior?

The browser on which the test will run, should receive the flags
### How would you reproduce the current behavior (if this is a bug)?

I'm trying to use Service Workers in the page, but they require `localhost` or `https` and _testcafe_ is currently redirecting the browser to a server of itself (therefore, not https neither localhost). I could send a couple flags to the browser and it would not be a problem:

```
testcafe chrome --ignore-certificate-errors --unsafely-treat-insecure-origin-as-secure=* tests/
```
### Specify your
- operating system: MacOS Sierra
- testcafe version: 0.9.0
- node.js version: 4.2.6

### PR DESCRIPTION
**Pull request by AndreyBelym**
\cc @inikulin @AlexanderMoskovkin 
### Feature summary
Allow to pass command line arguments for a browser starting through CLI. Example:

```
testcafe 'path:/usr/bin/chrome-browser --start-fullscreen' test.js 

testcafe 'path:`C:\Program Files (x86)\Google\Chrome\Application\chrome.exe` --start-fullscreen' test.js

testcafe 'chrome --start-fullscreen' test.js
```

### CODE REVIEW

**Comment by AndreyBelym** [comment]
FPR

**Comment by AndreyBelym** [comment]
Do not merge it, please, found a bug now, working on it!

File: test/server/browser-provider-test.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+        };
+    });
+
+    after(function () {
+        testcafeBrowserTools.getBrowserInfo   = originalBrowserToolsGetBrowserInfo;
+        testcafeBrowserTools.open             = originalBrowserToolsOpen;
+        testcafeBrowserTools.getInstallations = originalBrowserToolsGetInstallations;
+    });
+
+    it('path provider with arguments', function () {
```
Comment (AlexanderMoskovkin) [inline test/server/browser-provider-test.js]: `should parse path with arguments`

File: test/server/browser-provider-test.js [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        return getBrowserInfo('path:/usr/bin/chrome --arg1 --arg2')
+            .then(function (browserInfo) {
+                expect(browserInfo.path).to.be.equal('/usr/bin/chrome');
+                expect(browserInfo.cmd).to.be.equal('--internal-arg --arg1 --arg2');
+            });
+    });
+
+    it('path provider with spaces in path and arguments', function () {
```
Comment (AlexanderMoskovkin) [inline test/server/browser-provider-test.js]: `should parse path with spaces in path and arguments`

File: test/server/browser-provider-test.js [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+        return getBrowserInfo('path:`/opt/Google Chrome/chrome` --arg1 --arg2')
+            .then(function (browserInfo) {
+                expect(browserInfo.path).to.be.equal('/opt/Google Chrome/chrome');
+                expect(browserInfo.cmd).to.be.equal('--internal-arg --arg1 --arg2');
+            });
+    });
+
+    it('locally-installed provider with arguments', function () {
```
Comment (AlexanderMoskovkin) [inline test/server/browser-provider-test.js]: `should parse alias with arguments`

File: test/server/browser-provider-test.js (hunk 2 above)
Comment (MarinaRukavitsyna) [inline test/server/browser-provider-test.js]: should parse a path if there are spaces in a path name or arguments

Comment (AlexanderMoskovkin) [inline test/server/browser-provider-test.js]: `should parse browser parameters with arguments if there are spaces in a file path`

(approved by @MarinaRukavitsyna )

**Comment by AlexanderMoskovkin** [comment]
rebase required

**Comment by AndreyBelym** [comment]
FPR

File: test/server/browser-provider-test.js [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+        };
+    });
+
+    after(function () {
+        testcafeBrowserTools.getBrowserInfo   = originalBrowserToolsGetBrowserInfo;
+        testcafeBrowserTools.open             = originalBrowserToolsOpen;
+        testcafeBrowserTools.getInstallations = originalBrowserToolsGetInstallations;
+    });
+
+    it('Should parse path with arguments', function () {
```
Comment (MarinaRukavitsyna) [inline test/server/browser-provider-test.js]: it('Should parse browser parameters with arguments', function () {

**Comment by AndreyBelym** [comment]
FPR

**Comment by AndreyBelym** [comment]
\сс @MarinaRukavitsyna

**Pull request merged**